tests/lib/hardware/node_base.py

```diff
@@ -48,6 +48,13 @@ def get_ssh_ip(self) -> str:
         """
         pass
 
+    @abstractmethod
+    def add_data_disk(self, capacity):
```

Comment:
I think we need actually 3 methods:
* `disk_create`
* `disk_attach`
* `disk_detach`

to be able to create&attach new disks during a test. but that's something we could look into later I guess.
But could you rename the method to `disk_add` maybe?

Reply:
I agree. I'll add a TODO comment as it is not immediately necessary.